packages/astro/src/core/sync/index.ts

```
@@ -99,6 +110,9 @@ export async function sync(
 		}
 	} catch (e) {
 		const safeError = createSafeError(e);
+		if (isAstroError(e)) {
+			throw e;
+		}
```

Avoid swallowing AstroError hints and error codes